From: Iain Lane <iainl@gnome.org>
Date: Mon, 14 May 2018 22:32:25 +0200
Subject: libgdm: Drop weak refs on the GDBusConnection

The GDBusProxies hold a strong reference to the connection themselves,
so maintaining separate weak references is unnecessary.

This commit drops those extraneous weak references.

https://bugzilla.gnome.org/show_bug.cgi?id=795940
---
 libgdm/gdm-client.c | 47 -----------------------------------------------
 1 file changed, 47 deletions(-)

diff --git a/libgdm/gdm-client.c b/libgdm/gdm-client.c
index e7b93e3..f327344 100644
--- a/libgdm/gdm-client.c
+++ b/libgdm/gdm-client.c
@@ -652,13 +652,6 @@ gdm_client_open_reauthentication_channel_sync (GdmClient     *client,
                                                           cancellable,
                                                           error);
 
-        if (user_verifier != NULL) {
-                g_object_weak_ref (G_OBJECT (user_verifier),
-                                   (GWeakNotify)
-                                   g_object_unref,
-                                   connection);
-        }
-
  out:
         return user_verifier;
 }
@@ -763,12 +756,6 @@ gdm_client_get_user_verifier_sync (GdmClient     *client,
                 g_object_add_weak_pointer (G_OBJECT (client->priv->user_verifier),
                                            (gpointer *)
                                            &client->priv->user_verifier);
-                g_object_ref (connection);
-                g_object_weak_ref (G_OBJECT (client->priv->user_verifier),
-                                   (GWeakNotify)
-                                   g_object_unref,
-                                   client->priv->connection);
-
                 if (client->priv->enabled_extensions != NULL) {
                         gboolean res;
 
@@ -908,11 +895,6 @@ gdm_client_get_user_verifier_finish (GdmClient       *client,
                                    (gpointer *)
                                    &client->priv->user_verifier);
 
-        g_object_weak_ref (G_OBJECT (client->priv->user_verifier),
-                           (GWeakNotify)
-                           g_object_unref,
-                           client->priv->connection);
-
         return user_verifier;
 }
 
@@ -1081,10 +1063,6 @@ gdm_client_get_greeter_finish (GdmClient       *client,
                                    (gpointer *)
                                    &client->priv->greeter);
 
-        g_object_weak_ref (G_OBJECT (client->priv->greeter),
-                           (GWeakNotify)
-                           g_object_unref,
-                           client->priv->connection);
         return greeter;
 }
 
@@ -1129,10 +1107,6 @@ gdm_client_get_greeter_sync (GdmClient     *client,
                 g_object_add_weak_pointer (G_OBJECT (client->priv->greeter),
                                            (gpointer *)
                                            &client->priv->greeter);
-                g_object_weak_ref (G_OBJECT (client->priv->greeter),
-                                   (GWeakNotify)
-                                   g_object_unref,
-                                   client->priv->connection);
 
                 query_for_timed_login_requested_signal (client->priv->greeter);
         }
@@ -1263,11 +1237,6 @@ gdm_client_get_remote_greeter_finish (GdmClient     *client,
                                    (gpointer *)
                                    &client->priv->remote_greeter);
 
-        g_object_weak_ref (G_OBJECT (client->priv->remote_greeter),
-                           (GWeakNotify)
-                           g_object_unref,
-                           client->priv->connection);
-
         return remote_greeter;
 }
 
@@ -1311,11 +1280,6 @@ gdm_client_get_remote_greeter_sync (GdmClient     *client,
                 g_object_add_weak_pointer (G_OBJECT (client->priv->remote_greeter),
                                            (gpointer *)
                                            &client->priv->remote_greeter);
-                g_object_ref (connection);
-                g_object_weak_ref (G_OBJECT (client->priv->remote_greeter),
-                                   (GWeakNotify)
-                                   g_object_unref,
-                                   client->priv->connection);
         }
 
         return client->priv->remote_greeter;
@@ -1444,12 +1408,6 @@ gdm_client_get_chooser_finish (GdmClient       *client,
                                    (gpointer *)
                                    &client->priv->chooser);
 
-        g_object_ref (connection);
-        g_object_weak_ref (G_OBJECT (client->priv->chooser),
-                           (GWeakNotify)
-                           g_object_unref,
-                           client->priv->connection);
-
         return chooser;
 }
 
@@ -1493,11 +1451,6 @@ gdm_client_get_chooser_sync (GdmClient     *client,
                 g_object_add_weak_pointer (G_OBJECT (client->priv->chooser),
                                            (gpointer *)
                                            &client->priv->chooser);
-                g_object_ref (connection);
-                g_object_weak_ref (G_OBJECT (client->priv->chooser),
-                                   (GWeakNotify)
-                                   g_object_unref,
-                                   client->priv->connection);
         }
 
         return client->priv->chooser;
